### PR TITLE
fix: make gatsby id static for hot reloading

### LINKED ISSUE #44514: DX: hot-reloading challenges is broken.
## Affected page

All challenges in development.

## To reproduce

1. `npm run develop` in terminal
1. Go to localhost:8000/learn/responsive-web-design/basic-html-and-html5/say-hello-to-html-elements
2. Open curriculum/challenges/english/01-responsive-web-design/basic-html-and-html5/say-hello-to-html-elements.md in your editor
3. Change some text below --description--
4. See error in terminal
5. Do not see expected changes to the page

## Expected behavior

It should be possible to see changes in the development client without stopping and restarting `npm run develop`

## Additional context

This was caused by https://github.com/freeCodeCamp/freeCodeCamp/pull/44499

specifically the change to
(https://github.com/freeCodeCamp/freeCodeCamp/blob/main/client/plugins/fcc-source-challenges/create-challenge-nodes.js#L57)

The solution is *probably* to create a map between filename paths and the UUID that's generated for each path, but there might be a better approach.

### PR DESCRIPTION
**Pull request by moT01**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #44514

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This should do the trick, but I think it would be good to add a uniqueness check (in case of weirdness) and turn hot-reloading back.  It's not a lot of code and it means we can test the reloading.

**Comment by moT01** [comment]
Can you clarify what you mean by "turn hot-reloading back".

**Comment by ojeytonwilliams** [comment]
:man_facepalming: I wasn't thinking straight.  You can ignore that comment.

**Comment by moT01** [comment]
That part of the comment or the whole thing - do we still want the uniqueness check?

**Comment by ojeytonwilliams** [comment]
>  do we still want the uniqueness check?

I think so. It's *really* confusing when the ids clash, so it's worth guarding against that.

**Comment by cronlabspl** [comment]
> so it's worth guarding against that.

True.

**Comment by raisedadead** [comment]
To fix tests & update the status, when ready.

**Comment by cronlabspl** [comment]
> To fix tests & update the status, when ready.

Sorry, what?

**Review by ojeytonwilliams — APPROVED** [review]
@moT01 I realised how painful it was that this was broken, so I went ahead and added the tests.  I hope that's okay!

**Review by cronlabspl — APPROVED** [review]
Looks good to me 👍

**Review by cronlabspl — APPROVED** [review]
Looks good to me 👍

File: client/plugins/fcc-source-challenges/create-challenge-nodes.js [hunk 1]
```diff
@@ -54,7 +54,9 @@ function createChallengeNode(challenge, reporter) {
           sourceInstanceName: 'challenge'
         },
         { challenge },
-        { id: crypto.randomUUID() }
```
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/plugins/fcc-source-challenges/create-challenge-nodes.js:57]: This is part of the client build process, so won't effect the performance of the site.  If you do find it's slow, though, please open an issue so we can fix it.

**Review by ShaunSHamilton — APPROVED** [review]
LGTM 👍

**Pull request merged**